### PR TITLE
Migrate airflow-azure-data-factory-integration guide

### LINKED ISSUE #1029: Migrate airflow-azure-data-factory-integration
Migrate the existing guide airflow-azure-data-factory-integration (Executing Azure Data Factory Pipelines with Airflow) to Docs.

The branch for this is: airflow-azure-data-explorer-migration

### PR DESCRIPTION
**Pull request by kwcanuck**
Resolves #1029 

Migrate the existing guide airflow-azure-data-factory-integration (Executing Azure Data Factory Pipelines with Airflow) to Docs.

### CODE REVIEW

File: guides/airflow-azure-data-factory-integration.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+![ADF Role Assignment](/img/guides/adf_add_role_assignment2.png)
+
+Now you should be able to connect to your Data Factory from Airflow using your Client ID and Client Secret.
+
+Additional detail on requirements for interacting with Azure Data Factory using the REST API can be found [here](https://docs.microsoft.com/en-us/azure/data-factory/quickstart-create-data-factory-rest-api). You can also see [this link](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#register-an-application-with-azure-ad-and-create-a-service-principal) for more information on creating a registered application in Azure Active Directory
+
+### Create a DAG to orchestrate ADF jobs
+
+Now that our ADF pipelines are runnable, we create a DAG that executes those pipelines using Azure provider modules. The DAG below executes two ADF pipelines in parallel (tasks `run_pipeline1` and `run_pipeline2`). It then uses an `AzureDataFactoryPipelineRunStatusSensor` to wait until "pipeline2" has completed before finishing the DAG.
```
Comment (jwitz) [inline guides/airflow-azure-data-factory-integration.md:75]:
```suggestion
Now that our ADF pipelines are runnable, we create a DAG that executes those pipelines using Azure provider modules. The DAG below executes two ADF pipelines in parallel (tasks `run_pipeline1` and `run_pipeline2`). It then uses an `AzureDataFactoryPipelineRunStatusSensor` to wait until `pipeline2` has completed before finishing the DAG.
```

File: guides/airflow-azure-data-factory-integration.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+This example shows how to use ADF operators and sensors to orchestrate multiple ADF pipelines using Airflow.
+
+### ADF prerequisites
+
+Before you can orchestrate your ADF pipelines with Airflow, you have to make the pipelines runnable. 
+
+> **Note:** If you do not currently have an ADF pipeline in your Azure account and are new to ADF, check out the [ADF quick start docs](https://docs.microsoft.com/en-us/azure/data-factory/quickstart-create-data-factory-portal) for help getting started.
+
+To make your ADF pipeline accessible by Airflow you will need to register an App with Azure Active Directory to get a Client ID and Client Secret (API Key) for your Data Factory. First, go to Azure Active Directory and click **Registered Apps** to see a list of registered apps. If you created a Resource group, you should already have an app registered with the same name.
```
Comment (jwitz) [inline guides/airflow-azure-data-factory-integration.md:44]: This seems straightforward enough that I think we can turn it into a numbered process?

File: guides/airflow-azure-data-factory-integration.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
+---
+title: "Executing Azure Data Factory Pipelines with Airflow"
+sidebar_label: "Executing Azure Data Factory Pipelines with Airflow"
+description: "Triggering remote jobs in Azure Data Factory from your Apache Airflow DAGs."
```
Comment (jwitz) [inline guides/airflow-azure-data-factory-integration.md:4]:
```suggestion
description: "Trigger remote jobs in Azure Data Factory from your Apache Airflow DAGs."
```

**Review by jwitz — APPROVED** [review]
LGTM! Not required, but it might be nice to turn the primary integration information into numbered steps

File: guides/airflow-azure-data-factory-integration.md (hunk 2 above)
Comment (kwcanuck) [reply guides/airflow-azure-data-factory-integration.md:44]: @jwitz this doesn't seem like a task.

**Pull request merged**